sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -453,6 +494,47 @@ case class BroadcastHashJoinExec(
     val (keyEv, anyNull) = genStreamSideJoinKey(ctx, input)
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
+    val isLongHashedRelation = broadcastRelation.value.isInstanceOf[LongHashedRelation]
+
+    // fast stop if isOriginInputEmpty = true
+    // whether isNullAwareAntiJoin is true or false
+    // should accept all rows in streamedSide
+    if (broadcastRelation.value.isOriginInputEmpty) {
+      return s"""
+                |// Common Anti Join isOriginInputEmpty(true) accept all
+                |$numOutput.add(1);
+                |${consume(ctx, input)}
+          """.stripMargin
+    }
+
+    if (isNullAwareAntiJoin) {
+      if (broadcastRelation.value.allNullColumnKeyExistsInOriginInput) {
+        return s"""
+                  |// NAAJ isOriginInputEmpty(false) anyNullKeyExists(true) reject all
+            """.stripMargin
+      } else {
+        val found = ctx.freshName("found")
+        return s"""
+                  |// NAAJ isOriginInputEmpty(false) allNullColumnKeyExistsInOriginInput(false)
+                  |boolean $found = false;
+                  |// generate join key for stream side
+                  |${keyEv.code}
+                  |if (${ if (isLongHashedRelation) s"$anyNull" else s"${keyEv.value}.allNull()"}) {
```

Comment:
I am finding the `allNull` / `anyNull` to be mixed here. Why is one branch of the code checking for `anyNull`, and the other `allNull` ?

Is `allNull` in the else branch a holdover from "We may support multi-column one day" ?

Reply:
yes, it's for multi column, but i got you point about remain "single column" focussed
these allNull check should be in the next PR. ^_^

Reply:
Why is one branch of the code checking for anyNull, and the other allNull ?

because in LongHashedRelation
kevEv.value it a java long type, it certainly could not call allNull

as in UnsafeHashedRelation
kevEv.value is a UnsafeRow, this is why separately branch needed.

anyway, I will drop these code to remain "single column" focussed